### PR TITLE
function to check types and values of two numbers

### PR DESCRIPTION
**Pull request by prodxxx**

### CODE REVIEW

File: tests/strictequal.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      const secondValTest3 = 5
+
+      //call the function
+      let test3Result = strictEqual(firstValTest3, secondValTest3)
+
+      //make assertions that the tst passes
+      expect(test3Result).to.equal(true)
+    })
+  })
+})
```
Comment (jlcarmic) [inline tests/strictequal.test.js:52]: These tests should have been committed before the code as stated in the instructions.

File: tests/strictequal.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+const describe = require('mocha').describe
+const it = require('mocha').it
+const expect = require('chai').expect
+const strictEqual = require('../strictequal')
+
+describe('test 1 fail on parameter types types', () => {
+  describe('test 1 parameter are different types', () => {
```
Comment (jlcarmic) [inline tests/strictequal.test.js:7]: Remember that `describe` provides structure, you wouldn't have nested `describe`s for each test like this. Having a `describe` for the entire module, within that a `describe` for each function in the module, and then in that `it`s for each test is the preferred structure. This kind of breakdown makes it hard to read the code and the results.

File: tests/strictequal.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    it('test 1 types differ', () => {
```
Comment (jlcarmic) [inline tests/strictequal.test.js:8]: Your test descriptions should fully describe what is happening and expected. Something like `returns false when the parameters are the same value but different types` would be a lot clearer.

File: tests/strictequal.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      // create test data for fail test
+      const firstValTest1 = '5'
+      const secondValTest1 = 5
+
+      //call the function
+      let test1Result = strictEqual(firstValTest1, secondValTest1)
```
Comment (jlcarmic) [inline tests/strictequal.test.js:14]: Since this data is so simple, you can one line all of this rather than creating extra variables.

```JavaScript
const result = strictEqual('5', 5)
```

File: tests/strictequal.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+describe('test 2 fail on parameter values', () => {
+  describe('test 2 parameter are different values', () => {
+    it('test 2 values differ', () => {
+      // create test data for fail test
+      const firstValTest1 = 6
+      const secondValTest1 = 5
+
+      //call the function
+      let test1Result = strictEqual(firstValTest1, secondValTest1)
```
Comment (jlcarmic) [inline tests/strictequal.test.js:30]: This should be `const` since you do not reassign a value to this variable. Also you have `test1` variables in your test2. As a rule, you would never number tests.

File: tests/strictequal.test.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+const describe = require('mocha').describe
```
Comment (jlcarmic) [inline tests/strictequal.test.js:1]: Since strictEquals is about the same type and the same value I would expect 4 tests

1) Same type, same value (ex. 10, 10)
2) Same type, different value (ex. 10, 15)
3) Different type, same value (ex. 10, '10')
4) Different type, different value (ex. 10, '15')

File: strictequal.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const strictEqual = (value1, value2) => {
+  let value1type = typeof value1
+  let value2type = typeof value2
```
Comment (jlcarmic) [inline strictequal.js:3]: These should be `const` since you don't ever reassign a value to this variable

File: strictequal.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  if (value1type == value2type && value1 == value2) {
```
Comment (jlcarmic) [inline strictequal.js:4]: There's no need to create extra variable here you could just do the `typeof` right in the if condition

```JavaScript
if (typeof value1 == typeof value2 && value1 == value2) {
```

File: strictequal.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    return true
+  }
+  else {
+    return false
+  }
```
Comment (jlcarmic) [inline strictequal.js:9]: Here we are saying if the condition is true return true, if the condition is false return false. This means we can just return the condition since it will be the value we want returned.

```JavaScript
const strictEqual = (value1, value2) => {
    return typeof value1 == typeof value2 && value1 == value2
}
```

Also since then it would be a single line return we could use fat arrow's implicit return

```JavaScript
const strictEqual = (value1, value2) => typeof value1 == typeof value2 && value1 == value2
```

**Review by jlcarmic — APPROVED** [review]
You got to a working solution but there is a good amount of clean up that could happen here. Also you didn't follow the directions of commiting your tests first.

**Score:** 90%

**Explainer Video:**

File: tests/strictequal.test.js (hunk 6 above)
Comment (prodxxx) [reply tests/strictequal.test.js:1]: I only performed three tests because i am testing for different types first. Sine there are only two possibilities for the type test the minimum number of test to evaluate the accuracy is three.

File: strictequal.js (hunk 8 above)
Comment (prodxxx) [reply strictequal.js:4]: I was trying to do the type of in the if but couldn't seem to get the syntax right.  eslint kept barking at me.

File: tests/strictequal.test.js (hunk 6 above)
Comment (jlcarmic) [reply tests/strictequal.test.js:1]: Each of these four test types I laid out are necessary as otherwise you open up the possibility of edge cases creeping in. Different type, same value and different type, different value are distinct test cases and should both be covered.
